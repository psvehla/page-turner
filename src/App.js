import React, { Component } from 'react';

import { PAGE_ONE, PAGE_TWO, PAGE_THREE, PAGE_FOUR } from './pages.js';
import Header from './components/Header';
import Footer from './components/Footer';
import PageTurner from './components/PageTurner';

import './App.css';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      currentPage: PAGE_ONE,
    }
  }

  onClickHandler = async () => {
    switch (this.state.currentPage) {
      case PAGE_ONE:
        this.setState({
          currentPage: PAGE_TWO,
        });
        break;
        case PAGE_TWO:
          this.setState({
            currentPage: PAGE_THREE,
          });
          break;
          case PAGE_THREE:
            this.setState({
              currentPage: PAGE_FOUR,
            });
            break;
      default:
        this.setState({
          currentPage: PAGE_ONE,
        });
    }
  }

  render() {
    return (
      <div className="App">
        <Header />
        <PageTurner currentPage={this.state.currentPage} onClick={this.onClickHandler} />
        <Footer />
      </div>
    );
  }
}

export default App;
