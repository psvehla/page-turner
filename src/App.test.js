import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('renders correctly', () => {

  const tree = renderer
    .create(<App page="https://localhost:3000">AnimalSentences</App>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
