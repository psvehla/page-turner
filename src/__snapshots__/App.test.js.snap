// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header>
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
  </header>
  <div>
    <img
      alt="bike"
      onClick={[Function]}
      src="1.jpg"
    />
  </div>
  <footer>
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
  </footer>
</div>
`;
